drivers/sx1276/sx1276.c

```diff
@@ -0,0 +1,1302 @@
+                    xtimer_remove(&dev->_internal.rx_timeout_timer);
+
+                    sx1276_set_status(dev,  SX1276_RF_IDLE);
+
+                    send_event(dev, SX1276_RX_TIMEOUT);
+                    break;
+                default:
+                    break;
+            }
+            break;
+        case SX1276_RF_TX_RUNNING:
+            break;
+        default:
+            break;
+    }
+}
+
+void sx1276_on_dio2(void *arg)
+{
+    /* Get interrupt context */
+    sx1276_t *dev = (sx1276_t *) arg;
+
+    switch (dev->settings.state) {
+        case SX1276_RF_RX_RUNNING:
+            switch (dev->settings.modem) {
+                case SX1276_MODEM_LORA:
+                    if (dev->settings.lora.freq_hop_on) {
+                        /* Clear IRQ */
+                        sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGS, SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL);
+
+                        dev->_internal.last_channel = sx1276_reg_read(dev, SX1276_REG_LR_HOPCHANNEL) & SX1276_RF_LORA_HOPCHANNEL_CHANNEL_MASK;
+                        send_event(dev, SX1276_FHSS_CHANGE_CHANNEL);
+                    }
+
+                    break;
+                default:
+                    break;
+            }
+            break;
+        case SX1276_RF_TX_RUNNING:
+            switch (dev->settings.modem) {
+                case SX1276_MODEM_FSK:
+                    break;
+                case SX1276_MODEM_LORA:
+                    if (dev->settings.lora.freq_hop_on) {
+                        /* Clear IRQ */
+                        sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGS, SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL);
+
+                        dev->_internal.last_channel = sx1276_reg_read(dev, SX1276_REG_LR_HOPCHANNEL) & SX1276_RF_LORA_HOPCHANNEL_CHANNEL_MASK;
+                        send_event(dev, SX1276_FHSS_CHANGE_CHANNEL);
+                    }
+                    break;
+                default:
+                    break;
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+void sx1276_on_dio3(void *arg)
+{
+    /* Get interrupt context */
+    sx1276_t *dev = (sx1276_t *) arg;
+
+    switch (dev->settings.modem) {
+        case SX1276_MODEM_FSK:
+            break;
+        case SX1276_MODEM_LORA:
+            /* Clear IRQ */
+            sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGS, SX1276_RF_LORA_IRQFLAGS_CADDETECTED | SX1276_RF_LORA_IRQFLAGS_CADDONE);
+
+            /* Send event message */
+            dev->_internal.is_last_cad_success = (sx1276_reg_read(dev, SX1276_REG_LR_IRQFLAGS) & SX1276_RF_LORA_IRQFLAGS_CADDETECTED) == SX1276_RF_LORA_IRQFLAGS_CADDETECTED;
+            send_event(dev, SX1276_CAD_DONE);
+            break;
+        default:
+            break;
+    }
+}
+
+/* Following interrupt lines are not used */
+void sx1276_on_dio4(void *arg)
+{
+    (void) arg;
+}
+
+void sx1276_on_dio5(void *arg)
+{
+    (void) arg;
+}
+
+void *dio_polling_thread(void *arg)
```

Comment:
Hi @Cr0s!

Just read this discussion. 
Netdev2 allows a similar flow without the need of another thread on the driver side. For instance, the [enc28j60] driver calls netdev->event_callback with NETDEV2_ISR_EVENT and this [is translated to a message] to the gnrc_netdev2 thread. In the end, the isr function of the driver is called (as shown [here]).

In this case the dio_polling_thread could be the isr function of the driver (note the isr function is not in ISR context, the name is a bit tricky :P). Probably it would need some flags to identify which interrupt occured.

[enc28j60]:https://github.com/RIOT-OS/RIOT/blob/master/drivers/enc28j60/enc28j60.c#L219
[is translated to a message]:https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/netdev2/gnrc_netdev2.c#L59
[here]:https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/link_layer/netdev2/gnrc_netdev2.c#L141